Add elevated api key authentication

dalebradley opened the pull request.

macreman commented on interceptors/payment_authentication.go:
```diff
@@ -75,17 +68,39 @@ func (paymentAuthenticationInterceptor PaymentAuthenticationInterceptor) Payment
 		isGetRequest := http.MethodGet == r.Method
 		authUserIsPaymentCreator := authorisedUser == paymentSession.CreatedBy.ID
 		authUserHasPaymentLookupRole := helpers.IsRoleAuthorised(r, helpers.AdminPaymentLookupRole)
+		isApiKeyRequest := helpers.IsKeyElevatedPrivilegesAuthorised(r)
```
should this line be more like:

`isApiKeyRequest  := identityType == helpers.APIKeyIdentityType` ?

dalebradley replied:
Yep, good spot 👍

dalebradley commented:
Added unit tests for the elevated key changes and moved some of the code around to allow user details to be captured regardless of auth type used (oauth or key)
